src/asInput/index.jsx

```diff
@@ -67,6 +74,23 @@ const asInput = (WrappedComponent, inputType = undefined, labelFirst = true) =>
           value: nextProps.value,
         });
       }
+      if (nextProps.isValid !== this.props.isValid && !nextProps.validator) {
+        this.setState({
```

Comment:
If `nextProps.isValid`, `nextProps.validationMessage`, and `nextProps.validator` are all different and `!nextProps.validator` that could trigger up to 3 additional re-renders right?

Reply:
@jaebradley that's how React does it? Yeah I can consolidate by updating a dict then setting that as the state. I had assumed React would wait until the JS event loop got back.

Reply:
I think so? It's actually pretty unclear from [the docs what the exact batching logic looks like for `setState`](https://reactjs.org/docs/react-component.html#setstate) (something something I guess I *could* read the source-code)

> Think of setState() as a request rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.

> setState() does not always immediately update the component. It may batch or defer the update until later.

> setState() will always lead to a re-render unless shouldComponentUpdate() returns false. If mutable objects are being used and conditional rendering logic cannot be implemented in shouldComponentUpdate(), calling setState() only when the new state differs from the previous state will avoid unnecessary re-renders.

Reply:
Given the additional complexity these checks incur, I'm not really sure it's worth it. Also, why shouldn't it be allowed to initialize an input as invalid but still have a validator? Or, to have a validator function, but then override the valid state in a parent later?

I think that as long as the props are well-documented we should allow updating the isValid props even if there is a validator defined. It might also help to have two stories for InputText: one example to show using just `validator` synchronously , and a second example where validation is done in the parent asynchronously and only `isValid` and `validationMessage` are passed as props.

Reply:
I've edited the branch to bundle state changes in a staging dictionary.